Makefile.openssl

```diff
@@ -0,0 +1,9 @@
+# Exporting proper linking flags -- used by Makefile and tests/Makefile
+
+ifeq ($(shell uname),Darwin)
```

Comment:
Why do you only use OpenSSL on Mac?

Reply:
on Linux, `-lcrypto` works unconditionally; on OSX, OpenSSL is oftentimes installed via `brew` and the path to openssl includes and libraries is *NOT* on the default LD_LIBRARY_PATH and CFLAGS path, so on OSX (and on OSX only) I perform some "auto-detection" that attempts to find a suitable path to the OpenSSL install directory